vsphere/resource_vsphere_host_virtual_switch_test.go

```diff
@@ -0,0 +1,253 @@
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfig(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfigStandbyLink(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testAccResourceVSphereHostVirtualSwitchCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tp = t
+			resource.Test(t, tc.testCase)
+		})
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchPreCheck(t *testing.T) {
+	if os.Getenv("VSPHERE_HOST_NIC0") == "" {
+		t.Skip("set VSPHERE_HOST_NIC0 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_HOST_NIC1") == "" {
+		t.Skip("set VSPHERE_HOST_NIC1 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_ESXI_HOST") == "" {
+		t.Skip("set VSPHERE_ESXI_HOST to run vsphere_host_virtual_switch acceptance tests")
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		vars, err := testClientVariablesForResource(s, "vsphere_host_virtual_switch.switch")
+		if err != nil {
+			return errors.New("vsphere_host_virtual_switch.switch not found in state")
+		}
+
+		hsID, name, err := splitHostVirtualSwitchID(vars.resourceID)
+		if err != nil {
+			return err
+		}
+		ns, err := hostNetworkSystemFromHostSystemID(vars.client, hsID)
+		if err != nil {
+			return fmt.Errorf("error loading host network system: %s", err)
+		}
+
+		_, err = hostVSwitchFromName(vars.client, ns, name)
+		if err != nil {
+			if err.Error() == fmt.Sprintf("could not find virtual switch %s", name) && !expected {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return fmt.Errorf("expected vSwitch %s to be missing", name)
+		}
+		return nil
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchConfig() string {
+	return fmt.Sprintf(`
+variable "host_nic0" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "host_nic1" {
+  type    = "string"
+  default = "%s"
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = "%s"
+}
+
+data "vsphere_host" "esxi_host" {
+  name          = "%s"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.esxi_host.id}"
+
+  network_adapters = ["${var.host_nic0}", "${var.host_nic1}"]
+
+  active_nics  = ["${var.host_nic0}", "${var.host_nic1}"]
+  standby_nics = []
+}
+`, os.Getenv("VSPHERE_HOST_NIC0"), os.Getenv("VSPHERE_HOST_NIC1"), os.Getenv("VSPHERE_DATACENTER"), os.Getenv("VSPHERE_ESXI_HOST"))
```

Comment:
_minor_ 🤔 feels like it may be worth returning some kind of helper object from the testProvider code, rather than putting magic strings throughout the codebase?

Reply:
Yeah, I'd like to maybe take this offline as something to think about and maybe come up with something that we can do a re-factor push later on. Right now I think it's readable enough to scale reasonably well - personally longer term, once the provider becomes mature enough to support it, I'd love to do away with the vast majority of these variables and have the resources self-host these kinds of things if at all possible.